src/Core/Form/IdentifiableObject/CommandBuilder/Product/SEOCommandsBuilder.php

```diff
@@ -44,23 +47,55 @@ public function buildCommands(ProductId $productId, array $formData): array
 
         $seoData = $formData['seo'] ?? [];
         $redirectionData = $formData['seo']['redirect_option'] ?? [];
+
+        $commands = [];
         $command = new UpdateProductSeoCommand($productId->getValue());
+        $seoUpdate = false;
 
         if (isset($seoData['meta_title'])) {
             $command->setLocalizedMetaTitles($seoData['meta_title']);
+            $seoUpdate = true;
         }
         if (isset($seoData['meta_description'])) {
             $command->setLocalizedMetaDescriptions($seoData['meta_description']);
+            $seoUpdate = true;
         }
         if (isset($seoData['link_rewrite'])) {
             $command->setLocalizedLinkRewrites($seoData['link_rewrite']);
+            $seoUpdate = true;
         }
 
         if (isset($redirectionData['type'])) {
             $targetId = (int) ($redirectionData['target']['id'] ?? 0);
             $command->setRedirectOption($redirectionData['type'], $targetId);
+            $seoUpdate = true;
+        }
+
+        // Only return the command if some updates have been detected
+        if ($seoUpdate) {
+            $commands[] = $command;
+        }
+
+        if (isset($seoData['tags'])) {
+            if (!empty($seoData['tags'])) {
+                if (!is_array($seoData['tags'])) {
+                    throw new InvalidArgumentException('Expected tags to be a localized array');
+                }
+
+                $parsedTags = [];
+                foreach ($seoData['tags'] as $langId => $rawTags) {
+                    $parsedTags[$langId] = !empty($rawTags) ? explode(',', $rawTags) : [];
+                }
+
+                $commands[] = new SetProductTagsCommand(
+                    $productId->getValue(),
+                    $parsedTags
+                );
```

Comment:
Question out of scope
This setter method expects 2 arguments, and the new multi-shop command builder can handle only 1 argument at the moment.
I'm working now on the SEO command builder, and I already had to handle the case with `setRedirectOption` setter. I propose to pass extra arguments as a 4th parameter in command fields:
```php
        $config = new CommandBuilderConfig($this->modifyAllNamePrefix);
        $config
            // skipped code
            ->addMultiShopField('[seo][redirect_option][type]', 'setRedirectOption', CommandField::TYPE_STRING, [
                (int) ($formData['seo']['redirect_option']['target']['id'] ?? 0),
            ])
        ;
```
I doubt this trick could handle SEO tags here. What do you think about a more flexible implementation with a callback function ?
```php
        $config
            ->addMultiShopCallbackField(function (array $data, UpdateProductSeoCommand $command) {
               // Update the command here
            })
        ;
```

Reply:
@jolelievre
The "extra arguments" solution is already implemented in my current work in progress. I used a simple array, but you already know I don't like arrays :)))

I like a lot your idea of property pathes listed as a single string. Simple and intuitive to use. But it doesn't solve the type field definitons to provide as 3rd argument. I'm affraid this solution can't handle the most tricky cases, like the SEO tags here.
Using a callback function is pretty much like moving a piece of code from one class to another.